TESTING.md

```diff
@@ -102,7 +102,7 @@ You can test against a remote Datastore emulator as well.  To do this, set the `
   DatastoreOptions options = DatastoreOptions.newBuilder()
       .setProjectId("my-project-id") // must match project ID specified on remote machine
       .setHost("http://<hostname of machine>:<port>")
-      .setAuthCredentials(AuthCredentials.noAuth())
+      .setNoCredentials()
```

Comment:
If `credentials` is `null` in the `Builder` object  we attempt to do credentials discovery. We kind of need a special indicator for the "no credentials" case. This is the reason why we previously had `AuthCredentials.noAuth()` and now have the `setNoCredentials()` setter (and the private `noCredentials` boolean inside `ServiceOptions` and `ServiceOptions.Builder`).